ext/spl/spl_iterators.c

```diff
@@ -1390,6 +1390,30 @@ static inline int spl_cit_check_flags(zend_long flags)
 	return cnt <= 1 ? SUCCESS : FAILURE;
 }
 
+static zend_class_entry* get_ce_or_aggregate_ce(zval **zobject_dst, zval *zobject_buffer, int *inc_refcount)
+{
+	zend_class_entry *ce = NULL;
+
+	ce = Z_OBJCE_P(*zobject_dst);
+	if (!instanceof_function(ce, zend_ce_iterator)) {
+		if (instanceof_function(ce, zend_ce_aggregate)) {
+			zend_call_method_with_0_params(Z_OBJ_P(*zobject_dst), ce, &ce->iterator_funcs_ptr->zf_new_iterator, "getiterator", zobject_buffer);
+			if (EG(exception)) {
+				zval_ptr_dtor(zobject_buffer);
+				return NULL;
+			}
+			if (Z_TYPE_P(zobject_buffer) != IS_OBJECT || !instanceof_function(Z_OBJCE_P(zobject_buffer), zend_ce_traversable)) {
```

Comment:
Don't we need an actual Iterator here? Or else we'd have to keep recursively checking.